proto/penumbra/penumbra/core/app/v1alpha1/app.proto

```diff
@@ -9,13 +9,33 @@ import "penumbra/core/component/governance/v1alpha1/governance.proto";
 import "penumbra/core/component/ibc/v1alpha1/ibc.proto";
 import "penumbra/core/component/shielded_pool/v1alpha1/shielded_pool.proto";
 import "penumbra/core/component/stake/v1alpha1/stake.proto";
+import "penumbra/core/transaction/v1alpha1/transaction.proto";
 
 // Query operations for the overall Penumbra application.
 service QueryService {
   // Gets the app parameters.
   rpc AppParameters(AppParametersRequest) returns (AppParametersResponse);
+  // Returns the CometBFT transactions that occurred during a given block.
+  rpc TransactionsByHeight(TransactionsByHeightRequest) returns (stream TransactionsByHeightResponse);
 }
 
+// Requests the list of all transactions that occurred within a given block.
+message TransactionsByHeightRequest {
+  // The expected chain id (empty string if no expectation).
+  string chain_id = 1;
+  // The block heights to retrieve.
+  repeated uint64 block_heights = 2;
```

Comment:
What's the motivation for making this be streaming rather than unary? It seems like it might be simpler if it was just for a single height.

Reply:
There are two separate things going on:

1. the request type supports asking for multiple block heights at the same time; I thought this could be useful though only a single block height is supplied by pcli in this PR
2. the responses are streaming to reduce response gRPC message size

As for 1. since it's not being used it could be removed, but most of our existing APIs are streaming so 2. matches the existing pattern in our APIs